### PR TITLE
Add hasNoExtension to Path assertions

### LINKED ISSUE #2318: Add `hasNoExtension` to `Path` assertions
#### Example

```java
assertThat(Paths.get("file")).hasNoExtension();
```

Related to #2233.

### PR DESCRIPTION
**Pull request by Syemon**
#### Check List:
* Fixes #2318
* Unit tests : YES
* Javadoc with a code example (on API only) : YES
* PR meets the [contributing guidelines](https://github.com/assertj/assertj-core/blob/main/CONTRIBUTING.md)

Following the [contributing guidelines](https://github.com/assertj/assertj-core/blob/main/CONTRIBUTING.md) will make it easier for us to review and accept your PR.

### CODE REVIEW

File: src/main/java/org/assertj/core/api/AbstractPathAssert.java [hunk 1]
```diff
@@ -1936,4 +1936,28 @@ public SELF hasExtension(String expected) {
...
+  /**
+   * Verifies that the actual {@code Path} has no extension.
+   *
+   * <p>
+   * Example:
+   * <pre><code class='java'> Path pathWithoutExtension = Paths.get(&quot;file&quot;);
+   * Path pathWithExtension = Paths.get(&quot;file.txt&quot;);
+   *
+   * // assertion will pass
+   * assertThat(pathWithoutExtension).hasExtension();
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/AbstractPathAssert.java:1948]: hasNoExtension ;-)

File: src/main/java/org/assertj/core/api/AbstractPathAssert.java [hunk 2]
```diff
@@ -1936,4 +1936,28 @@ public SELF hasExtension(String expected) {
...
+   *
+   * // assertion will fail
+   * assertThat(pathWithExtension).hasExtension();</code></pre>
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/AbstractPathAssert.java:1951]: same

File: src/main/java/org/assertj/core/error/ShouldHaveNoExtension.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import java.nio.file.Path;
+
+/**
+ * Creates an error message indicating that a {@code Path} should have no extension.
+ */
+public class ShouldHaveNoExtension extends BasicErrorMessageFactory {
+
+  @VisibleForTesting
+  public static final String PATH_HAS_EXTENSION = "%nExpected actual path:%n  %s%n not to have an extension, but extension was:%n  %s";
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/error/ShouldHaveNoExtension.java:25]: no need to go to the next line which is used to align values to compare usually so `was: %s"` should be enough here.

File: src/main/java/org/assertj/core/error/ShouldHaveNoExtension.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+package org.assertj.core.error;
+
+import org.assertj.core.util.VisibleForTesting;
+
+import java.nio.file.Path;
+
+/**
+ * Creates an error message indicating that a {@code Path} should have no extension.
+ */
+public class ShouldHaveNoExtension extends BasicErrorMessageFactory {
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/error/ShouldHaveNoExtension.java:22]: add a unit test for this

**Review by joel-costigliola — CHANGES_REQUESTED** [review]
@Syemon nice work, just a few comments to address.

File: src/main/java/org/assertj/core/api/AbstractPathAssert.java (hunk 1 above)
Comment (Syemon) [reply src/main/java/org/assertj/core/api/AbstractPathAssert.java]: fixed

File: src/main/java/org/assertj/core/api/AbstractPathAssert.java (hunk 2 above)
Comment (Syemon) [reply src/main/java/org/assertj/core/api/AbstractPathAssert.java]: fixed

File: src/main/java/org/assertj/core/error/ShouldHaveNoExtension.java (hunk 3 above)
Comment (Syemon) [reply src/main/java/org/assertj/core/error/ShouldHaveNoExtension.java]: fixed

File: src/main/java/org/assertj/core/error/ShouldHaveNoExtension.java (hunk 4 above)
Comment (Syemon) [reply src/main/java/org/assertj/core/error/ShouldHaveNoExtension.java:22]: fixed

**Comment by joel-costigliola** [comment]
Integrated thanks @Syemon!

**Pull request closed without merging**